addons/pos_hr/models/hr_employee.py

```diff
@@ -35,3 +35,16 @@ def _unlink_except_active_pos_session(self):
                     error_msg += _("Employee: %s - PoS Config(s): %s \n") % (employee.name, ', '.join(config.name for config in config_ids))
 
             raise UserError(error_msg)
+
+    def _read(self, fields):
+        # as `pos_blackbox_be` is a certified module, it's hard to make fixes in it
+        # so this is a workaround to remove `insz_or_bis_number` field from the fields list
+        # as the parent hr.employee model will attempt to read it from hr.employee.public
+        # where it doesn't exist
+        if 'insz_or_bis_number' in fields:
+            pos_blackbox_be_installed = self.env['ir.module.module'].sudo().search_count([('name', '=', 'pos_blackbox_be'), ('state', '=', 'installed')])
+            had_hr_user_group = self.env.user.has_group('hr.group_hr_user')
+            if not pos_blackbox_be_installed or not had_hr_user_group:
+                fields.remove('insz_or_bis_number')
+
+        return super()._read(fields)
```

Comment:
Don't do this logic here. The problem is in the hr.employee.public so we patch there. Patching hr.employee comes with the risk of not being able to read 'insz_or_bis_number' field.

Reply:
@caburj thank you for the feedback, I moved the fix to hr.employee.public